src/Components/Components/test/Routing/RouteTableFactoryTests.cs

```diff
@@ -320,6 +407,44 @@ public void PrefersLiteralTemplateOverTemplateWithParameters()
             Assert.Null(context.Parameters);
         }
 
+        [Fact]
+        public void PrefersLiteralTemplateOverTemplateWithOptionalParameters()
+        {
+            // Arrange
+            var routeTable = new TestRouteTableBuilder()
+                .AddRoute("/users/1", typeof(TestHandler1))
+                .AddRoute("/users/{id?}", typeof(TestHandler2))
+                .Build();
+            var context = new RouteContext("/users/1");
+
+            // Act
+            routeTable.Route(context);
+
+            // Assert
+            Assert.NotNull(context.Handler);
```

Comment:
Also the reverse of the scenario that @pranavkm mentioned 👍  (reverse the optional and non-optional ones)

Reply:
@pranavkm and I discussed this on teams but I'll post notes on my understanding of how this should function (which I reflected in the tests).

For the situation above, Scenario 1 should actually match with TestHandler2 and Scenario 2 should match with TestHandler1.

IMO, it makes sense that if the routes are otherwise ambiguous but one has an optional parameter and the other does not, then it should favor the non-optional over the optional unless no parameter is provided.